### DISCUSSION TITLE
Can I create a custom parser supporting tree-sitter api?

### DISCUSSION THREAD

**Discussion by wrvsrx** [tree-sitter/tree-sitter#3476] [Grammars] [answered]
[djot](https://djot.net/) is hard to parse in the tree-sitter way because it's highly context-sensitive. However, it's very easy to write parser by hand (first line by line parsing then parse inline elements) since it doesn't need backtracking.

I was wondering if I can adapt a hand writting parser to tree-sitter api and pretend it's a tree-sitter parser? Although I lost the ability for incremental parsing, I can get accurate parsing result and adapt hand writting parser to other editors (such as neovim or helix).

If it's possible, can you give me some suggestions for how to do that?

**Comment by ObserverOfTime** [comment, marked as answer]
[tree-sitter-djot](https://github.com/treeman/tree-sitter-djot) is a thing. Alternatively, you can create your own split parser like [markdown](https://github.com/tree-sitter-grammars/tree-sitter-markdown).
